plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -6442,6 +6443,30 @@ public void testMergeSimpleQueryPartitioned()
         assertUpdate("DROP TABLE " + targetTable);
     }
 
+    @Test
+    public void testMergeWithSubselectWithoutHash()
```

Comment:
No need to add a new test.
Just change the existing tests where we enable the optimization manually to work around the bug. Search for references to the github issue in the codebase to find the call sites.

Reply:
Okay, I dropped this test + removed the `optimize_hash_generation` overrides in two tests marked with the issue number.